sw/host/opentitanlib/src/test_utils/object.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{Context, Result};
+use object::{Object, ObjectSection, ObjectSymbol};
+
+/// Load a symbol's data from an object file.
+pub fn symbol_data(object: &object::read::File, name: &str) -> Result<Vec<u8>> {
+    let mut symbols = object.symbols();
+    let symbol = symbols
+        .find(|symbol| symbol.name() == Ok(name))
+        .with_context(|| format!("could not find symbol {name} in ELF"))?;
+
+    let mut sections = object.sections();
+    let section = sections
+        .find(|section| Some(section.index()) == symbol.section_index())
+        .with_context(|| format!("could not find section containing {name}"))?;
+
+    let offset = (symbol.address() - section.address()) as usize;
+    let data = section.uncompressed_data()?;
+    let data = &data[offset..offset + symbol.size() as usize];
+
+    Ok(data.to_vec())
```

Comment:
I wonder whether we could return a slice, but I guess we would need a lifetime argument?

Reply:
No, the data has type `Cow` and if it's owned then you will be returning references to a local variable which is rejected by borrow checker.